sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2724,4 +2730,19 @@ class HiveDDLSuite
       }
     }
   }
+
+  test("SPARK-30785: create table like a partitioned table") {
```

Comment:
does this test fail without this patch? AFAIK `tracksPartitionsInCatalog` is true by default.

Reply:
Correct, the test fails without this patch.

Reply:
Check the PR description.

Reply:
This issue only exists in DS partitioned table.